Dockerfile.cpu

```diff
@@ -0,0 +1,61 @@
+FROM ubuntu:18.04 as build-stage
+
+WORKDIR /usr/src/app
+
+ENV LANG="C.UTF-8" LC_ALL="C.UTF-8" PATH="/opt/venv/bin:$PATH" PIP_NO_CACHE_DIR="false" DISPLAY=":0" CXXFLAGS="-I/opt/venv/lib/python3.6/site-packages/numpy/core/include"
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    python3 python3-pip python3-venv python3-dev build-essential zlib1g-dev libsdl2-dev libjpeg-dev \
+    wget nasm tar libbz2-dev libgtk2.0-dev cmake git libfluidsynth-dev libgme-dev \
+    libopenal-dev timidity libwildmidi-dev unzip libboost-all-dev libcanberra-gtk-module \
+    libopenmpi-dev libosmesa6-dev libgl1-mesa-glx libglfw3 && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY requirements.txt .
+
+RUN python3 -m venv /opt/venv && \
+    python3 -m pip install pip==19.3.1 pip-tools==4.2.0 wheel==0.33.6 setuptools==44.0.0
+
+RUN python3 -m piptools sync
+
+COPY . .
+
+FROM ubuntu:18.04 as dev-stage
```

Comment:
I don't understand the use of two stages here. Your second stage never uses artifacts produced in the first stage?

That is, I'm missing a `COPY --from=build-stage` below.

The idea with multi-stage builds usually is to compile and link everything you need in the build stage, then e.g. build a self-contained wheel. And in your last stage (which no longer contains compilers, linkers, etc.) you copy this wheel over from the build stage.

Reply:
yes this can go :)